FreeRTOS/Demo/CORTEX_MPU_M3_MPS2_QEMU_GCC/main.c

```diff
@@ -0,0 +1,82 @@
+/*
+ * FreeRTOS V202112.00
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * https://www.FreeRTOS.org
+ * https://github.com/FreeRTOS
+ *
+ */
+
+#include "app_main.h"
+#include <FreeRTOS.h>
+#include <task.h>
+
+#include <FreeRTOSConfig.h>
+
+#include <string.h>
+#include <stdarg.h>
+#include <stdio.h>
+
+void vApplicationIdleHook( void );
+void vApplicationTickHook( void );
+
+int main()
+{
+    app_main();
+    return 0;
+}
+
+/*-----------------------------------------------------------*/
+
+void vApplicationIdleHook( void )
```

Comment:
I understand why these are left blank as with configs enabled these are going off pretty frequently, but again, I think having some sort of message going off in these functions serves as a good learning experience for a user.

Reply:
messages in this function will be overwhelming, and will compromise a smooth functioning of the program

Reply:
Ya I definitely see what you are saying. Perhaps just a comment about what they do/how they tie in with the FreeRTOS config to give users a better idea of how things work - this could also be done in some of the other handler functions if you think that would be better.

Reply:
Or just disable these hooks in FreeRTOSConfig.h?

Reply:
I prefer to keep them, as this is a demo, and the user can see possibilities